src/game/NPCHandler.cpp

```diff
@@ -728,7 +756,17 @@ void WorldSession::HandleStableSwapPet(WorldPacket& recv_data)
 
     Pet* pet = _player->GetPet();
 
-    if (!pet || pet->getPetType() != HUNTER_PET)
+    if (!pet) // if pet not by your side, load from DB for a moment
```

Comment:
I do unsummon it again later. Unless you hack your outgoing packet, in which case you will lose pet.